### DISCUSSION TITLE
Welcome to aniMotum Discussions!

### DISCUSSION THREAD

**Discussion by ianjonsen** [ianjonsen/foieGras#28] [Announcements]
## 👋 Welcome!
  This is the place where aniMotum users can:
  * Ask questions.
  * Share ideas.
  * Share code snippets.
  * Share solutions to common problems encountered using aniMotum.

Discussions are meant to be informal, unlike raising an issue or feature request there's no particular format you need to follow.

If you have an issue that you think should be dealt with in the aniMotum code, then you can raise one [here](https://github.com/ianjonsen/foieGras/issues/new?assignees=&labels=&template=bug_report.md&title=) - please make sure you follow the format provided in the bug template!

**Comment by vheim** [comment]
Hello,

Currently, I am using aniMotum's fit_ssm() function to fit a correlated-random walk to Argos satellite tracking data. However, I have a question regarding the "min.dt" argument within the function and am not sure if I interpret this argument correctly. The documentation says: "minimum allowable time difference between observations; dt <= min.dt will be ignored by the SSM". 

My first question is: what is the unit of the time difference, i.e. minutes, hours, days?

My second question is: does that mean, that if I specify mind.dt = 10, if I have two consecutive detections that are less than e.g. 10 hours (?) apart they will be ignored? What is the intention of the min.dt argument - can it help to increase the goodness of model fit? If so, based on the literature it seems that SSMs have more issues with long time gaps, so would it be useful to have an argument that allows to specify long time gaps that should be excluded from the analysis?

Thank you in advance!

Vital

**Comment by ianjonsen** [comment]
Hi,

The unit is seconds. `min.dt` exists because some Argos data sets can have observations very close together in time (0 - 5 s), `min.dt` was intended to remove these cases, which can otherwise sometimes cause model convergence failures. So, not really a goodness-of-fit improvement. Generally, you shouldn't need to change from the default, `min.dt = 60`. 

Your point re: SSM having issues with long time gaps in the data is correct, however, I've left it for researchers to decide how they want to address that problem. For example, two simple approaches are: 1) split tracks based on some data gap duration criterion prior to fitting the SSM; 2) split tracks after fitting the SSM by removing SSM-predicted locations in the data gaps.

A future version of `aniMotum` will likely include an approach for handling data gaps.

Thanks for asking the question.

Ian

**Reply by vheim** [reply]
Hi Ian,

Perfect - thank you very much for your responses!

The min.dt argument makes sense now and I used the first suggestion, i.e. splitting up the track based on a gap duration threshold prior to fitting the SSM.

Best wishes,
Vital

**Comment by ghost** [comment]


**Reply by ianjonsen** [reply]
Hi, you can fit the model to multiple individuals in one call to `fit_ssm()` but, internally, this fits the model separately to each individual rather than simultaneously in a single, hierarchical model. So what you've done is fine.

**Comment by ghost** [comment]
Hi, I am wondering whether this package currently has the functionality to simulate tracks biased towards set environmental conditions (For example warmer temperatures?)

**Reply by ianjonsen** [reply]
Thanks for the question - it's something I've certainly thought about on multiple occasions. But, no, currently that is not possible with the simulation functions. It's not difficult to simulate such movement biases, but it is difficult to handle different spatially gridded environmental data sets in a generally applicable & relevant manner for as many species as possible. If there was enough interest - ie. enough people asking for this - some fairly simple tools could be introduced but this would be a year or two away.

**Comment by ianjonsen** [comment]
There's no simple way to do this if you're fitting the `mp` model to Argos data. You could extract the re-routed locations via `grab(fit, what = "r")` and then use `fit_mpm` to re-fit a move persistence model to the re-routed locations. This is not quite the same thing as fitting the `mp` model via `fit_ssm`, as it doesn't account for location uncertainty, so your mileage may vary.

**Comment by scifiona** [comment]
Hello Ian, 

I am trying to find the right commands for mapping individual tracks. So, instead of having them all together on the same map, I'd like to inspect the tracks of every animal (id) individually. 
In my dataset, there are 29 individuals, and it gets a little messy when it's all on the same map. I mean, it's great for an overview and also for the models, I think (because then they have more data to work with), but not if you want to check out the IDs individually. 

What I did was to use the grab function and then just ggplotted every ID with a for loop, but then I missed out on all the nice features in the animotum package.

Any idea how to code this inside of animotum?

**Reply by ianjonsen** [reply]
Hi, the simplest way to do this would be via `lapply` or with one of the `map` functions in the `purrr` package. Using `lapply`, the approach would be something like this:

```
fit_lst <- split(fit, fit$id)

lapply(fit_lst, function(x) aniMotum::map(x, what = "p", ...))
```

this will render individual maps for each predicted track. If you want them in a single file, then you could use the `pdf` function, eg:

```
pdf(file = "maps.pdf")
lapply(fit_lst, function(x) aniMotum::map(x, what = "p", ...))
dev.off()
```

Hope that helps. 

Ian

**Comment by axellegeounige** [comment]
Hello, 

I read that we could feed directly a dataframe into fit_mpm but I don't see any examples doesn't not include fit_ssm first. Would anyone knoow what the dataframe has to look like for the function to work without an ssm object ? 

Best

**Comment by ianjonsen** [comment]
Hi,

If you have a data.frame that is structured for `fit_ssm` (see `?format_data`) then you can use `fit_mpm` by supplying that data.frame. All you have to do is specify which columns contain the location coordinates, using the `coords` argument. For example, using the built in `ellie` data.frame, `coords = 4:5` indicates that the `lon`, `lat` variables are in columns 4 and 5 respectively:
``` 
fmp <- fit_mpm(ellie, model = "mpm", coords = 4:5)

plot(fmp)
```

Ian
